frontend/packages/console-dynamic-plugin-sdk/src/app/__tests__/AppInitSDK.spec.tsx

```diff
@@ -1,113 +1,90 @@
+
+    expect(mockProvider).not.toHaveBeenCalled();
+    expect(screen.getByText('Hello, OpenShift!')).toBeVisible();
   });
 
-  it('should wrap children with Provider', () => {
-    useReduxStoreSpy.mockImplementation(() => ({ store, storeContextPresent: false }));
-    const wrapper = shallow(
-      <AppInitSDK configurations={mockConfig}>
-        <div data-test-id="child-id">Hello!!</div>
-      </AppInitSDK>,
-    );
-    expect(wrapper.find(Provider)).toHaveLength(1);
-    expect(wrapper.find('[data-test-id="child-id"]')).toHaveLength(1);
+  it('should wrap children with a Provider if no store context is present', () => {
+    renderAppInitSDK(mockConfig, { store, storeContextPresent: false });
+
+    expect(mockProvider).toHaveBeenCalled();
+    expect(screen.getByText('Hello, OpenShift!')).toBeVisible();
   });
 
-  it('should call the hook useReduxStore', () => {
-    useReduxStoreSpy.mockImplementation(() => ({ store, storeContextPresent: true }));
-    shallow(
-      <AppInitSDK configurations={mockConfig}>
-        <div data-test-id="child-id">Hello!!</div>
-      </AppInitSDK>,
-    );
-    expect(useReduxStoreSpy).toHaveBeenCalled();
-    expect(useReduxStoreSpy).toHaveBeenCalledTimes(1);
+  it('should call the useReduxStore hook', () => {
+    renderAppInitSDK(mockConfig);
+
+    expect(useReduxStoreMock).toHaveBeenCalledTimes(1);
   });
 
-  it('should call the util setUtilsConfig with proper config', () => {
-    useReduxStoreSpy.mockImplementation(() => ({ store, storeContextPresent: true }));
-    mount(
-      <AppInitSDK configurations={mockConfig}>
-        <div data-test-id="child-id">Hello!!</div>
-      </AppInitSDK>,
-    );
-    expect(configSetupSpy).toHaveBeenCalled();
-    expect(configSetupSpy).toHaveBeenCalledTimes(1);
-    expect(configSetupSpy).toHaveBeenCalledWith({ appFetch: mockConfig.appFetch });
+  it('should call the setUtilsConfig utility with the proper config', () => {
+    renderAppInitSDK(mockConfig);
+
+    expect(configSetup.setUtilsConfig).toHaveBeenCalledTimes(1);
+    expect(configSetup.setUtilsConfig).toHaveBeenCalledWith({ appFetch: mockConfig.appFetch });
   });
 
-  it('should call apiDiscovery with store instance if provided and not default one', () => {
-    useReduxStoreSpy.mockImplementation(() => ({ store, storeContextPresent: true }));
-    const initApiDiscoverySpy = jest.spyOn(apiDiscovery, 'initApiDiscovery');
-    initApiDiscoverySpy.mockImplementation(jest.fn());
-    mount(
-      <AppInitSDK configurations={mockConfig}>
-        <div data-test-id="child-id">Hello!!</div>
-      </AppInitSDK>,
-    );
-    expect(mockConfig.apiDiscovery).toHaveBeenCalled();
-    expect(mockConfig.apiDiscovery).toHaveBeenCalledTimes(1);
-    expect(mockConfig.apiDiscovery).toHaveBeenCalledWith(store);
-    expect(initApiDiscoverySpy).toHaveBeenCalledTimes(0);
+  it('should call the provided apiDiscovery function if it exists', () => {
+    renderAppInitSDK(mockApiDiscoveryConfig);
```

Comment:
I think it's better to consistently explicitly define this config 

```suggestion
    renderAppInitSDK(mockApiDiscoveryConfig, { store, storeContextPresent: true });
```